pennylane/devices/tests/test_gates.py

```diff
@@ -79,11 +79,20 @@
     "DoubleExcitationPlus": qml.DoubleExcitationPlus(0, wires=[0, 1, 2, 3]),
     "DoubleExcitationMinus": qml.DoubleExcitationMinus(0, wires=[0, 1, 2, 3]),
     "QubitCarry": qml.QubitCarry(wires=[0, 1, 2, 3]),
-    "QubitSum:": qml.QubitSum(wires=[0, 1, 2]),
+    "QubitSum": qml.QubitSum(wires=[0, 1, 2]),
+    "PauliRot": qml.PauliRot(0, "XXYY", wires=[0, 1, 2, 3]),
+    "U1": qml.U1(0, wires=0),
+    "U2": qml.U2(0, 0, wires=0),
+    "U3": qml.U3(0, 0, 0, wires=0),
 }
 
 all_ops = ops.keys()
 
+# All qubit operations should be available to test in the device test suite
+all_available_ops = qml.ops._qubit__ops__.copy()  # pylint: disable=protected-access
```

Comment:
Should we also consider templates that are now operations? :thinking: Some templates could define `matrix` methods in future and be directly implementable/supported by devices.

Reply:
Good point. We need to rethink the device test suite. 

But I would say if a template is natively supported, this should be tested in the device?